catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/transformations/EmbeddedListMetacardsHandler.java

```diff
@@ -0,0 +1,112 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.metacard.workspace.transformations;
+
+import static java.util.stream.Collectors.toList;
+
+import com.google.common.annotations.VisibleForTesting;
+import ddf.action.ActionRegistry;
+import ddf.catalog.data.Metacard;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import org.codice.ddf.catalog.ui.metacard.workspace.ListMetacardImpl;
+import org.codice.ddf.catalog.ui.metacard.workspace.WorkspaceAttributes;
+import org.codice.ddf.catalog.ui.metacard.workspace.transformer.WorkspaceTransformer;
+import org.codice.ddf.configuration.SystemBaseUrl;
+
+public class EmbeddedListMetacardsHandler extends EmbeddedMetacardsHandler {
+
+  @VisibleForTesting static final String LIST_ACTION_PREFIX = "catalog.data.metacard.list";
+  @VisibleForTesting static final String ACTIONS_KEY = "actions";
+
+  private static final Set<String> EXTERNAL_LIST_ATTRIBUTES = Collections.singleton(ACTIONS_KEY);
+  private final ActionRegistry actionRegistry;
+
+  public EmbeddedListMetacardsHandler(ActionRegistry actionRegistry) {
+    super(WorkspaceAttributes.WORKSPACE_LISTS, ListMetacardImpl.TYPE);
+    this.actionRegistry = actionRegistry;
+  }
+
+  // Add "actions" key to list metacards.
+  @Override
+  public Optional<List> metacardValueToJsonValue(
+      WorkspaceTransformer transformer, List metacardXMLStrings, Metacard workspaceMetacard) {
+
+    final List<Map<String, Object>> listActions = getListActions(workspaceMetacard);
+
+    final Optional<List> listMetacardsOptional =
+        super.metacardValueToJsonValue(transformer, metacardXMLStrings, workspaceMetacard);
+
+    listMetacardsOptional.ifPresent(
+        listMetacards ->
+            ((List<Object>) listMetacards)
+                .stream()
+                .filter(Map.class::isInstance)
+                .map(Map.class::cast)
+                .forEach(listMetacardMap -> listMetacardMap.put(ACTIONS_KEY, listActions)));
+
+    return listMetacardsOptional;
+  }
+
+  // Remove "actions" key from list metacard map.
+  @Override
+  public Optional<List> jsonValueToMetacardValue(
+      WorkspaceTransformer transformer, List metacardJsonData) {
+
+    ((List<Object>) metacardJsonData)
+        .stream()
+        .filter(Map.class::isInstance)
+        .map(Map.class::cast)
+        .forEach(listMetacardMap -> removeExternalListAttributes(listMetacardMap));
+
+    return super.jsonValueToMetacardValue(transformer, metacardJsonData);
+  }
+
+  private void removeExternalListAttributes(Map listMetacardMap) {
+    EXTERNAL_LIST_ATTRIBUTES.forEach(listMetacardMap::remove);
+  }
+
+  /**
+   * Given a {@link org.codice.ddf.catalog.ui.metacard.workspace.WorkspaceMetacardImpl}, get a list
+   * of actions that can be executed on a list.
+   *
+   * @param workspaceMetacard
+   * @return
```

Comment:
✏️ If we're not going to fill out these annotations, why bother adding them?

Reply:
Removed